### DISCUSSION TITLE
All my NextJs API returns error 504

### DISCUSSION THREAD

**Discussion by robertagr** [vercel/community#3602] [Help]
### Summary

Locally everything works perfectly without error, but in Vercel I have this bunch of errors related to 504, every time I try to make a request to the database. I checked the Function Region both in Vercel and in my MongoDB and everything is settled correctly. What can be the issue here? 

This is my public repository: https://github.com/robertagr/SpotsHub/tree/main

Could you help me take a look and see what's going on? Thank you!

### Steps to Reproduce

<img width="601" alt="Screenshot 2023-08-14 at 12 44 21" src="https://github.com/vercel/community/assets/133139279/18537ce5-8b43-4b7a-ba78-ce1e791d5aff">

**Comment by amyegan** [comment]
I suspect the problem may be that the db files are not being included in the build the way they're expected in the code. You could move them to a `/src/utils` folder like in this example: https://nextjs.org/docs/getting-started/react-essentials#sharing-data-between-server-components

**Reply by robertagr** [reply]
Thank you for your answer. I just tried but unfortunately, I still have the same errors :(
